src/Microsoft.Performance.SDK/Processing/DataSourceAttribute.cs

```diff
@@ -6,32 +6,159 @@
 namespace Microsoft.Performance.SDK.Processing
 {
     /// <summary>
-    ///     Base attribute for denoting a data source that feeds into
-    ///     a custom data source. This class cannot be instantiated.
-    ///     This class is not meant to be derived by consumers.
+    ///     Base attribute for denoting a Data Source that feeds into
+    ///     a Custom Data Source. This class cannot be instantiated.
+    ///     <para/>
+    ///     Because users can implement their own
+    ///     <see cref="IDataSource"/>s, they can implement an
+    ///     Attribute that implements this class with their
+    ///     Data Source as the type parameter.
+    ///     <para />
+    ///     <example>
+    ///     This example shows a sample implementation:
+    ///     <code>
+    ///         [AttributeUsage(AttributeTargets.Class, /* properties elided */)]
```

Comment:
if derived classes no longer need `AttributeUsage`, remove this (and remove from docs 😄 )